app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -91,13 +95,41 @@ class PatientSearchResultItemView(
     addressLabel.text = address.completeAddress
   }
 
-  private fun renderPatientNameAgeAndGender(fullName: String, gender: Gender, dateOfBirth: DateOfBirth) {
+  private fun highlightPatientName(searchQuery: String, patientSearchResult: PatientSearchResultViewModel, dateOfBirth: DateOfBirth): Name {
+    val canHighlight = searchQuery.isNotBlank() && patientSearchResult.fullName.contains(searchQuery, ignoreCase = true)
+    return if (canHighlight) {
+      Name.Highlighted(
+          patientName = renderPatient(patientSearchResult.fullName, patientSearchResult.gender, dateOfBirth),
+          highlightStart = patientSearchResult.fullName.indexOf(searchQuery, ignoreCase = true),
+          highlightEnd = patientSearchResult.fullName.indexOf(searchQuery, ignoreCase = true) + searchQuery.length)
+    } else {
+      Name.Plain(renderPatient(patientSearchResult.fullName, patientSearchResult.gender, dateOfBirth))
+    }
+  }
+
+  private fun renderPatientNameAgeAndGender(name: Name) {
+    when (name) {
+      is Name.Highlighted -> {
+        val highlightName = SpannableStringBuilder(name.patientName)
+        val highlightColor = ContextCompat.getColor(context, R.color.facility_search_query_highlight)
```

Comment:
Let's create a separate color for this? or use `simple_light_blue_100`. Also, you can now start using the `context.resolveColor` method to get the color, theme attribute color.